tasks/main.yml

```
@@ -141,13 +153,17 @@
         firewall_config: "{{ __firewalld_facts.firewall_config }}"
 
 - name: Check config files after removal
-  when: __firewall_previous_replaced | bool
+  when:
+    - __firewall_previous_replaced | bool
+    - not ansible_check_mode
   block:
     - name: Get config files, checksums after
       script:
         cmd: >
           files/get_files_checksums.sh {{ __firewall_python_cmd | quote }}
-          {{ __firewall_firewalld_dir | quote }}
+          {{ __firewall_firewalld_dir | quote }} false
+          {{ __firewall_package_with_conf | quote }}
+          {{ __firewall_usr_lib_dir | quote }}
       register: __firewall_config_files_after
       changed_when: false
 
```

```suggestion
          {{ __firewall_firewalld_dir | quote }} false
          {{ __firewall_package_with_conf | quote }}
          {{ __firewall_usr_lib_dir | quote }}
      register: __firewall_config_files_after
      changed_when: false
      check_mode: false
```
Without removing check mode, the script will not run which will result in failure on task `Calculate what has changed`

---

I don't think using `check_mode: true` on the block is the same as running the playbook with `--check` - the problem is that I do not want the block starting at line 162 to be executed *at all* in check mode - this works correctly when running the playbook with `--check` - but when using `check_mode: true`, somehow `ansible_check_mode` is both true and false - it is False because the block is executed, but it is True because the script task complains . . .